cdc/sink/dmlsink/txn/event.go

```diff
@@ -51,21 +52,23 @@ func genTxnKeys(txn *model.SingleTableTxn) []uint64 {
 		return nil
 	}
 
-	hashRes := make(map[uint64]struct{}, len(txn.Rows))
+	keys := make([]uint64, 0, len(txn.Rows))
 	hasher := fnv.New32a()
 	for _, row := range txn.Rows {
+		if uint64(row.GetTableID()) > uint64(math.MaxUint32) {
+			log.Panic("TableID shouldn't be larger than math.MaxUint32")
+		}
+		prefix := uint64(row.GetTableID()) << 32
+
 		for _, key := range genRowKeys(row) {
 			if n, err := hasher.Write(key); n != len(key) || err != nil {
 				log.Panic("transaction key hash fail")
 			}
-			hashRes[uint64(hasher.Sum32())] = struct{}{}
```

Comment:
Why is the deduplication logic removed here?

Reply:
They are not necessary to be deduped now. So we can save this step to reduce CPU and memory usage.